### PR TITLE
chore: [GAC] Using correct permissions to fetch the application object using view mode

### PR DESCRIPTION
**Pull request by trishaanand**
Fixes https://github.com/appsmithorg/appsmith/issues/18979

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewPageServiceCEImpl.java [hunk 1]
```diff
@@ -358,15 +364,35 @@ public Mono<ApplicationPagesDTO> findApplicationPagesByApplicationIdViewModeAndB
...
+        if (view) {
+            permission = applicationPermission.getReadPermission();
+        } else {
+            permission = applicationPermission.getEditPermission();
+        }
 
-        return applicationService.findBranchedApplicationId(branchName, defaultApplicationId, applicationPermission.getReadPermission())
-                .flatMap(childApplicationId -> findApplicationPagesByApplicationIdViewMode(childApplicationId, view, markApplicationAsRecentlyAccessed))
+        return applicationService.findBranchedApplicationId(branchName, defaultApplicationId, permission)
+                .switchIfEmpty(Mono.error(new AppsmithException(AppsmithError.NO_RESOURCE_FOUND,
```
Comment (nsarupr) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewPageServiceCEImpl.java:375]: Can we make this `ACL_NO_RESOURCE_FOUND`?
Much more readable.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewPageServiceCEImpl.java [hunk 2]
```diff
@@ -358,15 +364,35 @@ public Mono<ApplicationPagesDTO> findApplicationPagesByApplicationIdViewModeAndB
...
+
+        AclPermission permission;
+        if (view) {
+            permission = applicationPermission.getReadPermission();
+        } else {
+            permission = applicationPermission.getEditPermission();
+        }
+
+        Mono<Application> applicationMono = applicationService.findByName(applicationName, permission)
                 .switchIfEmpty(Mono.error(new AppsmithException(AppsmithError.NO_RESOURCE_FOUND, FieldName.NAME, applicationName)))
```
Comment (nsarupr) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewPageServiceCEImpl.java:396]: Can we make this `ACL_NO_RESOURCE_FOUND`?
Much more readable.

**Review by nsarupr — APPROVED** [review]
Minor comments.
Otherwise looks good to me.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewPageServiceCEImpl.java (hunk 1 above)
Comment (trishaanand) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewPageServiceCEImpl.java:375]: Unfortunately, today the client side handling to redirect to 404 happens for NO_RESOURCE_FOUND. ACL_NO_RESOURCE_FOUND just throws a toast message. So we have to stick to this particular error code.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewPageServiceCEImpl.java (hunk 2 above)
Comment (trishaanand) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/NewPageServiceCEImpl.java:396]: Same as above.

**Pull request merged**